### PR TITLE
Update the `convertToRaw` script

### PR DESCRIPTION
**Pull request by fwyzard**
#### PR description:

Update the `convertToRaw` script:
  - change the default to 100 events per file, to match the value currently used online;
  - Improve the handling of the output and exit status of the subprocesses;
  - write a `run####_cff.py` file with the Source and Services to process the converted files;
  - add the `-r`/`--range` option to restrict the processing to a range of runs and lumisections;
  - add the `-v/--verbose` option to print additional message while preprocessing and converting the input files.

#### PR validation:

The new version runs and selects lumisections correctly.

Here is a sample output from running `convertToRaw` over one .root file:
```
$ convertToRaw -o /gpu_data/store/data/Run2022C/HLTPhysics/FED/v1 -r 34-65 root://eoscms.cern.ch//eos/cms/tier0/store/data/Run2022C/EphemeralHLTPhysics11/RAW/v1/000/357/329/00000/9e43f951-59ef-4dbf-85d1-28b5fa5f664f.root 
preprocessing input file root://eoscms.cern.ch//eos/cms/tier0/store/data/Run2022C/EphemeralHLTPhysics11/RAW/v1/000/357/329/00000/9e43f951-59ef-4dbf-85d1-28b5fa5f664f.root
  run 357329, lumisetion 29 is outside of the given range and will be skipped
  run 357329, lumisetion 30 is outside of the given range and will be skipped
  run 357329, lumisetion 31 is outside of the given range and will be skipped
  run 357329, lumisetion 32 is outside of the given range and will be skipped
  run 357329, lumisetion 33 is outside of the given range and will be skipped
  run 357329, lumisetion 34 with 463 events will be processed
  run 357329, lumisetion 35 with 431 events will be processed

found run 357329, lumi 34, with 463 events
cmsRun \
  /data/user/fwyzard/CMSSW_12_4_6/src/HLTrigger/Tools/python/convertToRaw.py \
  inputFiles=root://eoscms.cern.ch//eos/cms/tier0/store/data/Run2022C/EphemeralHLTPhysics11/RAW/v1/000/357/329/00000/9e43f951-59ef-4dbf-85d1-28b5fa5f664f.root \
  runNumber=357329 \
  lumiNumber=34 \
  eventsPerLumi=11655 \
  eventsPerFile=100 \
  outputPath=/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329_ls0034

found run 357329, lumi 35, with 431 events
cmsRun \
  /data/user/fwyzard/CMSSW_12_4_6/src/HLTrigger/Tools/python/convertToRaw.py \
  inputFiles=root://eoscms.cern.ch//eos/cms/tier0/store/data/Run2022C/EphemeralHLTPhysics11/RAW/v1/000/357/329/00000/9e43f951-59ef-4dbf-85d1-28b5fa5f664f.root \
  runNumber=357329 \
  lumiNumber=35 \
  eventsPerLumi=11655 \
  eventsPerFile=100 \
  outputPath=/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329_ls0035

```

And the corresponding `/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329_cff.py` file:
```python
import FWCore.ParameterSet.Config as cms

from EventFilter.Utilities.FedRawDataInputSource_cfi import source as _source
source = _source.clone(
    eventChunkSize = 200,   # MB
    eventChunkBlock = 200,  # MB
    numBuffers = 4,
    maxBufferedFiles = 4,
    fileListMode = True,
    fileNames = (
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0034_index000000.raw',
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0034_index000001.raw',
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0034_index000002.raw',
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0034_index000003.raw',
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0034_index000004.raw',
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0035_index000000.raw',
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0035_index000001.raw',
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0035_index000002.raw',
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0035_index000003.raw',
        '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1/run357329/run357329_ls0035_index000004.raw'
    )
)

from EventFilter.Utilities.EvFDaqDirector_cfi import EvFDaqDirector as _EvFDaqDirector
EvFDaqDirector = _EvFDaqDirector.clone(
    buBaseDir = '/gpu_data/store/data/Run2022C/HLTPhysics/FED/v1',
    runNumber = 357329
)

from EventFilter.Utilities.FastMonitoringService_cfi import FastMonitoringService as _FastMonitoringService
FastMonitoringService = _FastMonitoringService.clone()
```

#### If this PR will be backported please specify to which release cycle the backport is meant for:

To be backported to 12.4.x to convert data for use by the HLT.
See #39048.

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
please test

**Comment by missirol** [comment]
+hlt

File: HLTrigger/Tools/scripts/convertToRaw [hunk 1]
```diff
@@ -26,16 +26,54 @@ def cmsRun(config, **args):
...
+            (self.max_run == 0 or self.max_run == run) and (self.max_lumi == 0 or self.max_lumi >= lumi) or
+            (self.min_run != 0 and self.max_run > run)
+        )
+
+
 # default values
-events_per_file = 50
-events_per_lumi = 11650
+events_per_file = 100
+events_per_lumi = 11655
```
Comment (missirol) [inline HLTrigger/Tools/scripts/convertToRaw:68]: Just for my education: how was `11655` chosen?

Comment (fwyzard) [reply HLTrigger/Tools/scripts/convertToRaw:76]: That's how many events we can expect an HLT node to process in each lumisection under nominal conditions: `100 kHz / 200 nodes × 23.31 s = 11655 events`

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
Fixed the case where the output has a single .raw file.

**Comment by fwyzard** [comment]
please test

File: HLTrigger/Tools/scripts/convertToRaw [hunk 2]
```diff
@@ -199,7 +268,42 @@ for run in sorted(content):
...
         # write the final EoR file
         # implemented by hand instead of using json.dump() to match the style used by the DAQ tools
+        assert len(converted_files) == summary['data'][1]
         eor_file = run_path + f'/run{run:06d}_ls0000_EoR.jsn'
-        f = open(eor_file, 'w')
-        f.write('{\n   "data" : [ "%d", "%d", "%d", "%d" ],\n   "definition" : "%s",\n   "source" : "%s"\n}\n' % (summary['data'][0], summary['data'][1], summary['data'][2], summary['data'][3], summary['definition'], summary['source']))
-        f.close()
+        with open(eor_file, 'w') as file:
+            file.write('{\n   "data" : [ "%d", "%d", "%d", "%d" ],\n   "definition" : "%s",\n   "source" : "%s"\n}\n' % (summary['data'][0], summary['data'][1], summary['data'][2], summary['data'][3], summary['definition'], summary['source']))
+            file.close()
```
Comment (missirol) [inline HLTrigger/Tools/scripts/convertToRaw:275]: I thought with-open-as takes care of closing the file (so `close` wouldnt be needed).

File: HLTrigger/Tools/scripts/convertToRaw [hunk 3]
```diff
@@ -199,7 +268,42 @@ for run in sorted(content):
...
+from EventFilter.Utilities.EvFDaqDirector_cfi import EvFDaqDirector as _EvFDaqDirector
+EvFDaqDirector = _EvFDaqDirector.clone(
+    buBaseDir = '%s',
+    runNumber = %d
+)
+
+from EventFilter.Utilities.FastMonitoringService_cfi import FastMonitoringService as _FastMonitoringService
+FastMonitoringService = _FastMonitoringService.clone()
+""" % ('\n'.join("        '" + f + "'," for f in converted_files), args.output_directory, run))
+        file.close()
```
Comment (missirol) [inline HLTrigger/Tools/scripts/convertToRaw:307]: I thought with-open-as takes care of closing the file (so `close` wouldnt be needed).

**Comment by missirol** [comment]
+hlt

Review comment is minor.

File: HLTrigger/Tools/scripts/convertToRaw (hunk 2 above)
Comment (fwyzard) [reply HLTrigger/Tools/scripts/convertToRaw:275]: it does, but I don't mind doing it explicitly

**Comment by qliphy** [comment]
+1

**Pull request merged**